lib/api/graph.js

```diff
@@ -62,37 +62,48 @@ function snapshot(query, from, to, eventQuery, callback){
  *  };
  *  dogapi.initialize(options);
  *  var query = "system.cpu.idle{*}";
- *  var graph_json = {
- *         "viz": "timeseries",
- *         "requests": [
+ *  var graphJSON = {
+ *         viz: "timeseries",
+ *         requests: [
  *           {
- *             "q": query,
- *             "aggregator": "avg",
- *             "conditional_formats": [],
- *             "type": "area"
+ *             q: query,
+ *             aggregator: "avg",
+ *             conditional_formats: [],
+ *             type: "area"
  *           }
  *         ]
  *       }
- *  var timeframe = '1_hour';
- *  var size = 'xlarge';
- *  var legend = 'yes';
- *  var title = 'my awesome embed';
- *  dogapi.graph.createEmbed(graph_json, timeframe, size, legend, title, function(err, res){
+ *  var timeframe = "1_hour";
+ *  var size = "xlarge";
+ *  var legend = "yes";
+ *  var title = "my awesome embed";
+ *  dogapi.graph.createEmbed(graphJSON, timeframe, size, legend, title, function(err, res){
  *    console.dir(res);
  *  });
  *  ```
  */
-function createEmbed(graph_json, timeframe, size, legend, title, callback){
+function createEmbed(graphJSON, timeframe, size, legend, title, callback){
     var lastArgument = arguments.length -1;
 
+    // Build the form body
+    var body = "graph_json=" + encodeURIComponent(JSON.stringify(graphJSON));
+    if (timeframe !== undefined) {
+        body += "&timeframe=" + encodeURIComponent(timeframe);
+    }
+    if (size !== undefined) {
```

Comment:
Maybe we should just accept the full formObject and callback. That way we push this type of logic onto the user and then all of these do not have to be required and things are way cleaner here. And can just urlEncode the whole payload and pass it to datadog directly. 

so maybe something like 
```
createEmbed(embedJSON, callback)
```